cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/workflow/SparkProgramWorkflowRunner.java

```diff
@@ -63,20 +65,21 @@
 
   /**
    * Executes given {@link Program} with the given {@link ProgramOptions} and block until it completed.
-   * On completion, return the {@link RuntimeContext} of the program.
+   * On completion, currently this always returns the null.
    *
    * @throws Exception if execution failed.
    */
   @Override
-  public RuntimeContext runAndWait(Program program, ProgramOptions options) throws Exception {
+  public WorkflowToken runAndWait(Program program, ProgramOptions options) throws Exception {
     ProgramController controller = programRunnerFactory.create(ProgramRunnerFactory.Type.SPARK).run(program, options);
 
     if (controller instanceof SparkProgramController) {
-      final RuntimeContext context = ((SparkProgramController) controller).getContext();
-      return executeProgram(controller, context);
+      executeProgram(controller, ((SparkProgramController) controller).getContext());
     } else {
       throw new IllegalStateException("Failed to run program. The controller is not an instance of " +
                                         "SparkProgramController");
     }
+    // TODO: currently Spark program returns the null token
+    return null;
```

Comment:
Returning `null` here, since currently the `WorkflowToken` interface only has single method `getMapReduceCounters` in it which is specific to the MapReduce programs.